plumbing/object/tree.go

```diff
@@ -297,9 +297,10 @@ func (iter *treeEntryIter) Next() (TreeEntry, error) {
 
 // TreeWalker provides a means of walking through all of the entries in a Tree.
 type TreeWalker struct {
-	stack     []treeEntryIter
+	stack     []*treeEntryIter
```

Comment:
This seems safer to me; I was worried things that accessed the stack might end up calling `Next` on a copy, rather than the real object.  But the way the stack is used now, I don't think it's a real bug.  If you want me to revert it, let me know.

Reply:
I personally don't mind the change, the struct is just a pointer and an int so copies are not expensive and it might be faster due to locality but I don't think it matters much.